### PR TITLE
CHET-620: Run final FS migration and validation page

### PR DESCRIPTION
**Pull request by nanux**
* start and monitor DB and final FS migration
* validation page
* destroy the migration stack

### CODE REVIEW

File: frontend/src/main/dc-migration-assistant-fe/components/shared/MigrationTransferPageActions.tsx [hunk 1]
```diff
@@ -70,14 +70,16 @@ const ActionButton: FunctionComponent<{
...
+    testId?: string;
+}> = ({ loading, onClick, text, isRefresh, testId }) => (
     <Button
         style={{
             padding: '5px',
         }}
         onClick={onClick || ((): void => undefined)}
         isLoading={loading}
         appearance={isRefresh ? 'default' : 'primary'}
+        testId={testId || `button-${text.replace(/\W/g, '-').toLowerCase()}`}
```
Comment (nanux) [inline frontend/src/main/dc-migration-assistant-fe/components/shared/MigrationTransferPageActions.tsx:82]: Added some testability to react.

File: frontend/src/main/dc-migration-assistant-fe/components/stage/Validation.tsx [hunk 2]
```diff
@@ -124,7 +124,7 @@ const ValidationSummary: FunctionComponent = () => {
                 )}
                 id="finish-api-error"
             />
-            <h3>{I18n.getText('atlassian.migration.datacenter.step.validation.phrase')}</h3>
+            <h1>{I18n.getText('atlassian.migration.datacenter.step.validation.phrase')}</h1>
```
Comment (nanux) [inline frontend/src/main/dc-migration-assistant-fe/components/stage/Validation.tsx:127]: Consistency with other pages.

File: jira-e2e-tests/cypress/support/pages/ValidationPage.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+export const showsValidationPage = () => {
+    cy.location().should((loc: Location) => {
+        expect(loc.pathname).to.contain('/plugins/servlet/dc-migration-assistant/validation');
+    });
+    cy.get('#dc-migration-assistant-root h1').contains('Step 7 of 7: Validation');
+    cy.get('#dc-migration-assistant-root p > a')
+        .contains('http://')
+        .then((href) => {
+            cy.log(href === undefined ? '' : href.attr('href'));
```
Comment (nanux) [inline jira-e2e-tests/cypress/support/pages/ValidationPage.ts:9]: Here we get the service URL.

File: jira-e2e-tests/cypress/support/pages/FinalSync.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        'Now that youve blocked user access to the instance, we can copy its database and sync any new content changes. You can close this page and return anytime to check its progress.'
+    );
+
+    cy.get('button[data-testid=button-cancel').should('be.visible');
+    cy.get('button[data-testid=button-sync-now-').contains('Sync now!').click();
+};
+
+export const monitorFinalSync = (ctx: AppContext) => {
+    cy.get('#dc-migration-assistant-root h1').contains('Step 6 of 7: Final Sync');
+    cy.get('#dc-migration-assistant-root h4').contains('Database export', { timeout: 20000 });
```
Comment (nanux) [inline jira-e2e-tests/cypress/support/pages/FinalSync.ts:14]: These timeouts are sometimes flaky - I have a hunch that it is due to a race condition when we get the status via cypress test and it returns `CREATE_COMPLETE` but the UI is still not reflected. I am investigating it now.

File: frontend/src/main/dc-migration-assistant-fe/components/stage/Validation.tsx (hunk 2 above)
Comment (bpar476) [inline frontend/src/main/dc-migration-assistant-fe/components/stage/Validation.tsx:127]: Why is this changed to `h1` and what does it look like now?

File: jira-e2e-tests/cypress/integration/e2e_full_migration.ts [hunk 5]
```diff
@@ -100,6 +100,17 @@ describe('Migration plugin', () => {
...
 
     it.skip('show warning to block access access', () => {
-        showsBlockUserWarning(ctx);
+        showsBlockUserWarning();
+        continueWithMigration();
+        runFinalSync();
+        monitorFinalSync(ctx);
+    });
+
+    it.skip('runs final sync', () => {
```
Comment (bpar476) [inline jira-e2e-tests/cypress/integration/e2e_full_migration.ts:109]: It looks like we're trying to run final sync twice here. Was this just for development purposes?

File: jira-e2e-tests/cypress/support/pages/FinalSync.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { waitForStatus } from '../waiters';
+export const runFinalSync = () => {
+    cy.get('#dc-migration-assistant-root h1').contains('Step 6 of 7: Final Sync');
+    cy.get('#dc-migration-assistant-root p').contains(
+        'Now that youve blocked user access to the instance, we can copy its database and sync any new content changes. You can close this page and return anytime to check its progress.'
+    );
+
+    cy.get('button[data-testid=button-cancel').should('be.visible');
+    cy.get('button[data-testid=button-sync-now-').contains('Sync now!').click();
```
Comment (bpar476) [inline jira-e2e-tests/cypress/support/pages/FinalSync.ts:9]: Is this meant to have a trailing `-`?

File: jira-e2e-tests/cypress/support/pages/LandingPage.ts [hunk 7]
```diff
@@ -1,5 +1,4 @@
 export const startMigration = (ctx: AppContext) => {
-    cy.visit(ctx.pluginHomePage);
```
Comment (bpar476) [inline jira-e2e-tests/cypress/support/pages/LandingPage.ts:2]: Is this not needed because the test starts with `cy.visit(ctx.pluginFullUrl)`?

File: jira-e2e-tests/cypress/integration/e2e_full_migration.ts [hunk 8]
```diff
@@ -100,6 +100,17 @@ describe('Migration plugin', () => {
     });
 
     it.skip('show warning to block access access', () => {
```
Comment (dylanrathbone) [inline jira-e2e-tests/cypress/integration/e2e_full_migration.ts:102]: nit, test has the word `access` in it twice

File: jira-e2e-tests/cypress/support/pages/ValidationPage.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    });
+
+    cy.get('button').should('contain.text', 'Close the migration app').and('be.disabled');
+
+    cy.get('input[type=checkbox]')
+        .should('not.be.checked')
+        .click({ force: true })
+        .should('be.checked');
+
+    cy.get('button').should('contain.text', 'Close the migration app').and('be.enabled').click();
```
Comment (bpar476) [inline jira-e2e-tests/cypress/support/pages/ValidationPage.ts:27]: Is there any way we can check that the new window is opened or does cypress block the pop-up?

File: jira-e2e-tests/cypress/integration/e2e_full_migration.ts (hunk 8 above)
Comment (nanux) [reply jira-e2e-tests/cypress/integration/e2e_full_migration.ts:102]: Thanks, this was for development purposes, I will remove the full test.

File: frontend/src/main/dc-migration-assistant-fe/components/stage/Validation.tsx (hunk 2 above)
Comment (nanux) [reply frontend/src/main/dc-migration-assistant-fe/components/stage/Validation.tsx:127]: This was done to make it consistent with the rest of the pages where we use `<h1>` for the page title. I have a test running locally that I don't want to interrupt but I will paste the screenshot here.

I think it was just omission before as the design was inconsistent.

File: jira-e2e-tests/cypress/integration/e2e_full_migration.ts (hunk 5 above)
Comment (nanux) [reply jira-e2e-tests/cypress/integration/e2e_full_migration.ts:109]: Yes, this is skipped and my plan was to remove all fragmented test cases from the e2e test.

File: jira-e2e-tests/cypress/support/pages/FinalSync.ts (hunk 6 above)
Comment (nanux) [reply jira-e2e-tests/cypress/support/pages/FinalSync.ts:9]: It is, we are producing the `data-testid` with this function `testId={testId || `button-${text.replace(/\W/g, '-').toLowerCase()}`}`

File: jira-e2e-tests/cypress/support/pages/LandingPage.ts (hunk 7 above)
Comment (nanux) [reply jira-e2e-tests/cypress/support/pages/LandingPage.ts:2]: Yes, we are navigating to plugin home with `before()` hook within the test.

File: jira-e2e-tests/cypress/support/pages/ValidationPage.ts (hunk 9 above)
Comment (nanux) [reply jira-e2e-tests/cypress/support/pages/ValidationPage.ts:27]: I will check if that is possible. I haven't seen any popup from the cypress test though.

File: frontend/src/main/dc-migration-assistant-fe/components/stage/Validation.tsx (hunk 2 above)
Comment (nanux) [reply frontend/src/main/dc-migration-assistant-fe/components/stage/Validation.tsx:127]: image

File: frontend/src/main/dc-migration-assistant-fe/components/shared/MigrationTransferPageActions.tsx (hunk 1 above)
Comment (nanux) [reply frontend/src/main/dc-migration-assistant-fe/components/shared/MigrationTransferPageActions.tsx:82]: This is following the Atlaskit API - https://atlaskit.atlassian.com/packages/design-system/button

File: jira-e2e-tests/cypress/support/pages/ValidationPage.ts (hunk 9 above)
Comment (nanux) [reply jira-e2e-tests/cypress/support/pages/ValidationPage.ts:28]: - [ ] Investigate whether we can verify the popup
